### ISSUE TITLE
separation average

### ISSUE THREAD

**Issue by adematti** [manodeep/Corrfunc#271]
Currently the separation average in each bin (thetaavg, ravg, savg, rpavg) is computed *without* taking weights into account.
I think, for usual purposes, we would like to include weights (when provided) in this computation?
Typically, we would expect the same average separation, whether a galaxy is weighted by 0 or is totally removed from the catalog.
It would always be possible, for someone truly interested in the non-weighted separation average, to run the computation without weights.
What do you think?

Best

**Comment by lgarrison** [comment]
Yes, I think you're right! Everything should be computed using the weights. Would you like to open a PR?

**Comment by manodeep** [comment]
@adematti Do you need the weighted average separations for your Science case or is this more of a consistency/implementation choice question?

@lgarrison We had a long discussion about this when you first started implementing the weights within Corrfunc. We came to the conclusion that it was better to leave the average separations as is - since that would be the most likely use-case. And that if any user required a weighted separation, then they could "easily" add a custom weight function and keep track of the weighted mean separation.

**Comment by adematti** [comment]
@manodeep well, I'd think the weighted version would be the most intuitive convention for the separation average ; this is the definition we will use in DESI --- though we will most certainly make sure cosmological results do not depend on this choice.
In case one'd not want the weighted-average version, it would be straightforward to recompute the pair counts without weights.

Anyway, it's just fine to have this change only in the branch we are currently using in DESI.

**Comment by manodeep** [comment]
Alright great - as always I prefer to add a user-option that allows either weighted or unweighted average separations. Would the plan be the same - provide the default PAIR_PRODUCT weighted average for separations? And anything else the user has to code those up?

Regardless, if we do change the behaviour, we should try to keep backwards compatibility by default. That way this is not a breaking change but rather a functionality update.

What do y'all think?

**Comment by lgarrison** [comment]
Yeah, we already have an `output_ravg` field that we normally set to `True`, but we could allow the user to set that to the string `'weighted'` instead. Then the average separation would just inherit whatever weighting scheme the user is using.

**Comment by adematti** [comment]
To me this seems like a good idea.
I had weighted average separation implemented upon the current master version in https://github.com/adematti/Corrfunc/tree/separation-avg. Yet, this branch does not allow for the option that you suggest for output_ravg and some things would need to be modified if https://github.com/manodeep/Corrfunc/pull/258 is accepted.
